src/dodal/devices/fast_grid_scan.py

```diff
@@ -181,181 +196,165 @@ def non_integer_dwell_time(cls, dwell_time_ms: float) -> float:
         return dwell_time_ms
 
 
-class GridScanCompleteStatus(DeviceStatus):
+class PandAGridScanParams(GridScanParamsCommon):
     """
-    A Status for the grid scan completion
-    A special status object that notifies watchers (progress bars)
-    based on comparing device.expected_images to device.position_counter.
+    Params for panda constant-motion scan. Adds on the goniometer run-up distance
     """
 
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-        self.start_ts = time.time()
-
-        self.device.position_counter.subscribe(self._notify_watchers)
-        self.device.status.subscribe(self._running_changed)
-
-        self._name = self.device.name
-        self._target_count = self.device.expected_images.get()
-
-    def _notify_watchers(self, value, *args, **kwargs):
-        if not self._watchers:
-            return
-        time_elapsed = time.time() - self.start_ts
-        try:
-            fraction = 1 - value / self._target_count
-        except ZeroDivisionError:
-            fraction = 0
-            time_remaining = 0
-        except Exception as e:
-            fraction = None
-            time_remaining = None
-            self.set_exception(e)
-            self.clean_up()
-        else:
-            time_remaining = time_elapsed / fraction
-        for watcher in self._watchers:
-            watcher(
-                name=self._name,
-                current=value,
-                initial=0,
-                target=self._target_count,
-                unit="images",
-                precision=0,
-                fraction=fraction,
-                time_elapsed=time_elapsed,
-                time_remaining=time_remaining,
+    run_up_distance_mm: float = 0.17
+
+    def get_param_positions(self):
+        param_positions = super().get_param_positions()
+        param_positions["run_up_distance_mm"] = self.run_up_distance_mm
+        return param_positions
+
+
+class MotionProgram(Device):
+    def __init__(self, prefix: str, name: str = "") -> None:
+        super().__init__(name)
+        self.running = epics_signal_r(float, prefix + "PROGBITS")
+        self.program_number = epics_signal_r(float, prefix + "CS1:PROG_NUM")
+
+
+class ExpectedImages(SignalR[int]):
+    def __init__(self, parent: "FastGridScanCommon") -> None:
+        super().__init__(SoftSignalBackend(int))
+        self.parent = parent
+
+    async def get_value(self):
+        x = int(await self.parent.x_steps.get_value())  # type: ignore
+        y = int(await self.parent.y_steps.get_value())  # type: ignore
+        z = int(await self.parent.z_steps.get_value())  # type: ignore
+        first_grid = x * y
+        second_grid = x * z
+        return first_grid + second_grid
```

Comment:
Should: I think if you specify the type of the parent you can fix these type issues:
```suggestion
        self.parent: "FastGridScanCommon" = parent

    async def get_value(self):
        x = await self.parent.x_steps.get_value()
        y = await self.parent.y_steps.get_value()
        z = await self.parent.z_steps.get_value()
        first_grid = x * y
        second_grid = x * z
        return first_grid + second_grid
```